plugins/destination_testing.go

```diff
@@ -3,243 +3,225 @@ package plugins
-		Time:   time.Now(),
-		Status: schema.Present,
-	}
-	jsonColumn := &schema.JSON{
-		Bytes:  []byte(`{"test": "test"}`),
-		Status: schema.Present,
-	}
-	uuidArrayColumn := &schema.UUIDArray{}
-	if err := uuidArrayColumn.Set([]string{"00000000-0000-0000-0000-000000000001", "00000000-0000-0000-0000-000000000002"}); err != nil {
-		panic(err)
-	}
-	inetColumn := &schema.Inet{}
-	if err := inetColumn.Set("192.0.2.1/24"); err != nil {
-		panic(err)
-	}
-	inetArrayColumn := &schema.InetArray{}
-	if err := inetArrayColumn.Set([]string{"192.0.2.1/24", "192.0.2.1/24"}); err != nil {
-		panic(err)
-	}
-	cidrColumn := &schema.CIDR{}
-	if err := cidrColumn.Set("192.0.2.1"); err != nil {
-		panic(err)
-	}
-	cidrArrayColumn := &schema.CIDRArray{}
-	if err := cidrArrayColumn.Set([]string{"192.0.2.1", "192.0.2.1"}); err != nil {
-		panic(err)
-	}
-	macaddrColumn := &schema.Macaddr{}
-	if err := macaddrColumn.Set("aa:bb:cc:dd:ee:ff"); err != nil {
-		panic(err)
-	}
-	macaddrArrayColumn := &schema.MacaddrArray{}
-	if err := macaddrArrayColumn.Set([]string{"aa:bb:cc:dd:ee:ff", "11:22:33:44:55:66"}); err != nil {
-		panic(err)
-	}
-
-	data := schema.CQTypes{
-		cqID,
-		cqParentID,
-		boolColumn,
-		intColumn,
-		floatColumn,
-		uuidColumn,
-		textColumn,
-		byteaColumn,
-		textArrayColumn,
-		intArrayColumn,
-		timestampColumn,
-		jsonColumn,
-		uuidArrayColumn,
-		inetColumn,
-		inetArrayColumn,
-		cidrColumn,
-		cidrArrayColumn,
-		macaddrColumn,
-		macaddrArrayColumn,
-	}
-
-	return data
+func getTestLogger(t *testing.T) zerolog.Logger {
+	t.Helper()
+	zerolog.TimeFieldFormat = zerolog.TimeFormatUnixMs
+	return zerolog.New(zerolog.NewTestWriter(t)).Output(
+		zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: time.StampMicro},
+	).Level(zerolog.DebugLevel).With().Timestamp().Logger()
+}
+
+func (s *destinationTestSuite) destinationPluginTestWriteOverwrite(ctx context.Context, p *DestinationPlugin, logger zerolog.Logger, spec specs.Destination) error {
+	spec.WriteMode = specs.WriteModeOverwrite
+	if err := p.Init(ctx, logger, spec); err != nil {
+		return fmt.Errorf("failed to init plugin: %w", err)
+	}
+	tableName := "cq_test_write_overwrite"
+	table := testdata.TestTable(tableName)
+	syncTime := time.Now().UTC()
+	tables := []*schema.Table{
+		table,
+	}
+	if err := p.Migrate(ctx, tables); err != nil {
+		return fmt.Errorf("failed to migrate tables: %w", err)
+	}
+
+	sourceName := uuid.NewString()
+	resource := schema.DestinationResource{
+		TableName: table.Name,
+		Data:      testdata.TestData(),
+	}
+	resource2 := schema.DestinationResource{
+		TableName: table.Name,
+		Data:      testdata.TestData(),
+	}
+	_ = resource2.Data[5].Set("00000000-0000-0000-0000-000000000007")
+	resources := []schema.DestinationResource{
+		resource,
+		resource2,
+	}
+
+	if s.tests.DeleteStale {
+		if err := p.DeleteStale(ctx, tables, sourceName, syncTime); err != nil {
+			return fmt.Errorf("failed to delete stale data: %w", err)
+		}
+	}
+
```

Comment:
```suggestion
```

I mean you can remove this block and the test still passes so it seems redundant

Reply:
Oh I see there were two delete stales. The first one is a redundant yes.